app/scripts/migrations/078.js

```diff
@@ -0,0 +1,59 @@
+import { cloneDeep } from 'lodash';
+import { v4 } from 'uuid';
+
+const version = 78;
+
+/**
+ * Migrate the frequentRpcListDetail from the PreferencesController to the NetworkController, convert it from an array to an object
+ * keyed by random uuids, and update property `nickname` to `chainName`.
+ */
+export default {
+  version,
+  async migrate(originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData);
+    versionedData.meta.version = version;
+    const state = versionedData.data;
+    const newState = transformState(state);
+    versionedData.data = newState;
+    return versionedData;
+  },
+};
+
+function transformState(state) {
+  const { PreferencesController, NetworkController } = state || {};
+
+  if (!PreferencesController?.frequentRpcListDetail) {
```

Comment:
Wonder if it's worth it to return early if `PreferencesController` doesn't exist? That way we don't have to say `PreferencesController?.foo`, `PreferencesController?.bar`, etc.

Reply:
done [here](https://github.com/MetaMask/metamask-extension/pull/17421/commits/14e17a724380a89fc3155c6a45dc5ee9a5847f7a)